metaserver/src/main/scala/scala/meta/languageserver/Semanticdbs.scala

```diff
@@ -1,12 +1,39 @@
 package scala.meta.languageserver
 
 import java.nio.file.Files
+import scala.meta.interactive.InteractiveSemanticdb
+import scala.meta.languageserver.compiler.ScalacProvider
+import scala.meta.parsers.ParseException
 import com.typesafe.scalalogging.LazyLogging
 import org.langmeta.internal.semanticdb.schema.Database
 import org.langmeta.io.AbsolutePath
 import scala.meta.semanticdb
+import scala.util.Try
+import org.langmeta.inputs.Input
 
 object Semanticdbs extends LazyLogging {
+  def toSemanticdb(
+      input: Input.VirtualFile,
+      scalac: ScalacProvider
+  ): Option[semanticdb.Database] =
+    for {
+      path <- Uri.unapply(input.path)
+      compiler <- scalac.getCompiler(path)
+      document <- Try {
+        InteractiveSemanticdb.toDocument(
+          compiler = compiler,
+          code = input.value,
+          filename = input.path,
+          timeout = 10000
+        )
+      }.fold({
+        case _: ParseException => None // expected
+        case err =>
+          logger.error(s"Failed to emit semanticdb for $path", err)
+          None
+      }, Option.apply)
+    } yield semanticdb.Database(document.copy(language = "Scala212") :: Nil)
```

Comment:
what's the default language of `document` here? `"Scala211"`?

Reply:
I think it may be empty string actually because it was failing to parse, it doesn't really matter what we put here though